tidy3d/plugins/autograd/invdes/penalties.py

```diff
@@ -1,86 +1,140 @@
 from typing import Tuple, Union
 
 import autograd.numpy as np
+from numpy.typing import NDArray
 
+from tidy3d.components.base import Tidy3dBaseModel
 from tidy3d.components.types import ArrayFloat2D
 
 from ..types import PaddingType
-from .parametrizations import make_filter_and_project
+from .parametrizations import FilterAndProject
 
 
-def make_erosion_dilation_penalty(
-    filter_size: Tuple[int, ...],
-    beta: float = 100.0,
-    eta: float = 0.5,
-    delta_eta: float = 0.01,
-    padding: PaddingType = "reflect",
-):
-    """Computes a penalty for erosion/dilation of a parameter map not being unity.
-
-    Accepts a parameter array normalized between 0 and 1. Uses filtering and projection methods
-    to erode and dilate the features within this array. Measures the change in the array after
-    eroding and dilating (and also dilating and eroding). Returns a penalty proportional to the
-    magnitude of this change. The amount of change under dilation and erosion is minimized if
-    the structure has large feature sizes and large radius of curvature relative to the length scale.
+class ErosionDilationPenalty(Tidy3dBaseModel):
+    """A class that computes a penalty for erosion/dilation of a parameter map not being unity.
 
     Parameters
     ----------
-    filter_size : Tuple[int, ...]
-        The size of the filter to be used for erosion and dilation.
-    beta : float, optional
-        Strength of the tanh projection. Default is 100.0.
-    eta : float, optional
-        Midpoint of the tanh projection. Default is 0.5.
-    delta_eta : float, optional
-        The binarization threshold for erosion and dilation operations. Default is 0.01.
-    padding : PaddingType, optional
-        The padding type to use for the filter. Default is "reflect".
-
-    Returns
-    -------
-    Callable
-        A function that computes the erosion/dilation penalty for a given array.
+    filtproj : FilterAndProject
```

Comment:
hm, this is an interesting approach.. but maybe we need a good default value for this based on the default penalty params?

Reply:
Ah you mean not having the parametrization here be a customizable thing but built into the penalty? We can do that, wasn't sure which way is best.

Reply:
yea or (similar comment above) we can make the parameters (beta, eta, etc.) be the pd.Fields here.

Reply:
with good defaults